website/project/model.py

```diff
@@ -950,6 +937,11 @@ def pk(self):
         return self._id
 
     @property
+    def logs(self):
+        """ List of logs associated with this node"""
+        return list(NodeLog.find(Q('node', 'eq', self._id)).sort('date'))
```

Comment:
This shouldn't be wrapped in `list`.
Calling `list` on a query object loads the entire query into memory, which will get very slow very fast.
returning just `NodeLog.find(...).sort('date')` won't load any logs into memory until they are actually being used and it supports a list like interface. (Minus negative indexing)
And calling `len` on `Node.logs` will run a count query rather than count the list that is loaded.

Negative indexing can be mocked but just return the query sorted on `'-date'`